client/faucet/src/main.rs

```diff
@@ -0,0 +1,339 @@
+    /// To keep Testnet simple, we used one private key for both treasury compliance account and testnet
+    /// designated dealer account, hence here we only accept one private key.
+    /// To manually generate a keypair, use generate-key:
+    /// `cargo run -p generate-keypair -- -o <output_file_path>`
+    #[structopt(short = "m", long, default_value = "/opt/libra/etc/mint.key")]
+    pub mint_key_file_path: String,
+    /// Chain ID of the network this client is connecting to.
+    /// For mainnet: \"MAINNET\" or 1, testnet: \"TESTNET\" or 2, devnet: \"DEVNET\" or 3, \
+    /// local swarm: \"TESTING\" or 4
+    /// Note: Chain ID of 0 is not allowed; Use number if chain id is not predefined.
+    #[structopt(short = "c", long, default_value = "2")]
+    pub chain_id: libra_types::chain_id::ChainId,
+}
+
+#[tokio::main]
+async fn main() {
+    let args = Args::from_args();
+    libra_logger::Logger::new().init();
+
+    let address: std::net::SocketAddr = format!("{}:{}", args.address, args.port)
+        .parse()
+        .expect("invalid address or port number");
+
+    info!(
+        "[faucet]: chain id: {}, server url: {}",
+        args.chain_id,
+        args.server_url.as_str(),
+    );
+    let service = std::sync::Arc::new(mint::Service::new(
+        args.server_url,
+        args.chain_id,
+        args.mint_key_file_path,
+    ));
+
+    info!("[faucet]: running on: {}", address);
+    warp::serve(routes(service)).run(address).await;
+}
+
+fn routes(
+    service: std::sync::Arc<mint::Service>,
+) -> warp::filters::BoxedFilter<(impl warp::Reply,)> {
+    // POST /mint?amount=25&auth_key=xxx&currency_code=XXX
+    let mint = warp::path("mint")
+        .and(warp::post())
+        .and(warp::any().map(move || std::sync::Arc::clone(&service)))
+        .and(warp::query().map(move |params: mint::MintParams| params))
+        .and_then(handle)
+        .with(warp::log::custom(|info| {
+            info!(
+                "{} \"{} {} {:?}\" {} \"{}\" \"{}\" {:?}",
+                OptFmt(info.remote_addr()),
+                info.method(),
+                info.path(),
+                info.version(),
+                info.status().as_u16(),
+                OptFmt(info.referer()),
+                OptFmt(info.user_agent()),
+                info.elapsed(),
+            )
+        }));
+
+    let health = warp::path!("-" / "healthy").map(|| "libra-faucet:ok");
+    health.or(mint).boxed()
+}
+
+async fn handle(
+    service: std::sync::Arc<mint::Service>,
+    params: mint::MintParams,
+) -> Result<Box<dyn warp::Reply>, warp::Rejection> {
+    let ret = service.process(&params).await;
+    match ret {
+        Ok(body) => Ok(Box::new(body)),
+        Err(err) => Err(warp::reject::custom(ServerInternalError(err.to_string()))),
+    }
+}
+
+struct OptFmt<T>(Option<T>);
+
+impl<T: fmt::Display> fmt::Display for OptFmt<T> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        if let Some(ref t) = self.0 {
```

Comment:
if you don't want to write `Some(ref t) = self.0` you can also write `Some(t) = &self.0`